docs/api/collections.rst

```diff
@@ -183,3 +183,228 @@ Deleting a collection
                 "last_modified": 1434642842010
             }
         }
+
+
+Collection JSON schema
+======================
+
+A `JSON schema <http://json-schema.org/>`_ can optionally be associated to a
+collection.
+
+Once a schema is set, records will be validated during creation or update.
+
+If the validation fails, a ``400 Bad Request`` error response will be
+returned.
+
+.. note::
+
+    JSON schema is quite verbose and not an ideal solution for every use-case.
+    However it is universal and supported by many programming languages
+    and environments.
+
+
+Set or replace a schema
+-----------------------
+
+Just modify the ``schema`` attribute of the collection object:
+
+**Example request**
+
+.. code-block:: bash
+
+    $ echo '{
+      "data": {
```

Comment:
After thinking this through it might make sense to put it in data since it is an attribute of the collection.